clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractHeartbeatRequestManager.java

```diff
@@ -378,11 +378,17 @@ private void onErrorResponse(final R response, final long currentTimeMs) {
             case INVALID_REQUEST:
             case GROUP_MAX_SIZE_REACHED:
             case UNSUPPORTED_ASSIGNOR:
-            case UNSUPPORTED_VERSION:
                 logger.error("{} failed due to {}: {}", heartbeatRequestName(), error, errorMessage);
                 handleFatalFailure(error.exception(errorMessage));
                 break;
 
+            case UNSUPPORTED_VERSION:
+                message = "The cluster doesn't yet support the new consumer group protocol." +
+                        " Set group.protocol=classic to revert to the classic protocol until the cluster is upgraded.";
+                logger.error("{} failed due to {}: {}", heartbeatRequestName(), error, message);
```

Comment:
Does the `handleFatalFailure()` method bubble this up to the user? If so, do we need to log the error here too?

Reply:
Yes, we give user message in the exception (the assertion in test case), so we don't need to log it here. Thanks.